org.apache.aries.typedevent.bus/src/main/java/org/apache/aries/typedevent/bus/impl/EventConverter.java

```diff
@@ -296,35 +297,41 @@ public Map<String, Object> toUntypedEvent() {
     }
 
     @SuppressWarnings("unchecked")
-	public <T> T toTypedEvent(TypeData targetEventClass) {
-        Class<?> rawType = targetEventClass.getRawType();
-        Type genericTarget = targetEventClass.getType();
-		if (rawType.isInstance(originalEvent) && rawType == genericTarget) {
-			if(_log.isDebugEnabled()) {
-    			_log.debug("No need to convert {} to {} as it is already an instance", originalEvent, rawType);
-    		}
-            return (T) originalEvent;
-        } else {
-        	if(custom != null) {
-        		if(_log.isDebugEnabled()) {
-        			_log.debug("Using custom converter to convert {} to {}", originalEvent, genericTarget);
-        		}
-				Object result = custom.toTypedEvent(originalEvent, rawType, genericTarget);
-				if(result != null) {
-					return (T) result;
+	public <T> T toTypedEvent(TypeData typeData) {
+		Class<?> rawType = typeData.getRawType();
+		Type genericTarget = typeData.getType();
+		if (rawType.isInstance(originalEvent) && rawType == genericTarget && !typeData.isForceConversion()) {
```

Comment:
```suggestion
		if (rawType.isInstance(originalEvent) && rawType == genericTarget) {
```
We check `isForceConversion()` a few lines further on. Checking it here makes that check unnecessary, when I think it is intended to be checked.

Reply:
Another, possibly better, option would be to have

```java
if(typeDate.isForceConversion()) {
	if (_log.isDebugEnabled()) {
		_log.debug("Conversion is forced by property {}", TypeData.ARIES_EVENT_FORCE_CONVERSION);
	}
} else if (rawType.isInstance(originalEvent) && rawType == genericTarget && !typeData.isForceConversion()) {
	...
}
```